src/active_handout_plugins/templating.py

```diff
@@ -0,0 +1,77 @@
+from markdown.postprocessors import Postprocessor
+from markdown.preprocessors import Preprocessor
+from jinja2 import Environment
+import random
+import string
+
+
+class RandomIntVariable:
+    def __init__(self):
+        self.new()
+
+    def new(self, start=0, end=10):
+        self.value = random.randint(start, end)
+        return self.value
+
+    def __str__(self):
+        return str(self.value)
+
+
+class RandomFloatVariable:
+    def __init__(self):
+        self.new()
+
+    def new(self, start=0, end=10):
+        sz = end - start
+        self.value = sz * random.random() + start
+        return self.value
+
+    def __str__(self):
+        return str(self.value)
+
+
+class RandomStringVariable:
+    def __init__(self):
+        self.new()
+
+    def new(self, sz=10):
+        self.value = ''.join([random.choice(string.ascii_letters) for _ in range(sz)])
+        return self.value
+
+    def __str__(self):
+        return str(self.value)
+
+
+class Chooser:
+    def __init__(self):
+        pass
+
+    def __call__(self, *args):
+        return random.choice(args)
+
+
+class Jinja2PreProcessor(Preprocessor):
+    def __init__(self, md, user_provided_variables):
+        super().__init__(md)
+        self.user_provided_variables = user_provided_variables
+
+    def run(self, lines):
+        text = '\n'.join(lines)
+        e = Environment(extensions=['jinja2.ext.do'])
+        custom_template_values = {
+            'choose': Chooser(),
+            'seed': random.seed,
+        }
+        custom_template_values.update(
```

Comment:
Does this mean we are limited to 10 random int variables?

My suggestion is to create a custom dict overriding __getitem__:

```python
class TemplateValuesDict(UserDict):
    def __getitem__(self, key):
        if key.startswith('randint'):
            # add if doesnt exist
```

Reply:
Tried this and it doesn't work in jinja2. Don't know why though